prism_compile.c

```diff
@@ -2790,8 +2790,9 @@ pm_add_ensure_iseq(LINK_ANCHOR *const ret, rb_iseq_t *iseq, int is_return, const
 static void
 pm_insert_local_index(pm_constant_id_t constant_id, int local_index, st_table *index_lookup_table, rb_ast_id_table_t *local_table_for_iseq, pm_scope_node_t *scope_node)
 {
-    ID local = pm_constant_id_lookup(scope_node, constant_id);
-    local_table_for_iseq->ids[local_index] = local;
+    if (constant_id != 0) {
+        local_table_for_iseq->ids[local_index] = pm_constant_id_lookup(scope_node, constant_id);
+    }
     st_insert(index_lookup_table, constant_id, local_index);
```

Comment:
This is going to insert `0` into the index lookup table for every local that is anonymous. I _think_ this means that code like:

```ruby
def foo(*, &); end
```

will insert them both into the lookup table. Is there a way we could skip the `pm_insert_local_index` call entirely?